azure/durable_functions/orchestrator.py

```diff
@@ -135,12 +141,17 @@ def _add_to_actions(self, generation_state):
         if self.durable_context.will_continue_as_new:
             return
         if not generation_state._is_yielded:
-            if (isinstance(generation_state, Task)
-                    and hasattr(generation_state, "action")):
-                self.durable_context.actions.append([generation_state.action])
-            elif (isinstance(generation_state, TaskSet)
-                    and hasattr(generation_state, "actions")):
-                self.durable_context.actions.append(generation_state.actions)
+            if isinstance(generation_state, Task):
+                if self.durable_context._replay_schema == ReplaySchema.V1:
+                    self.durable_context.actions.append([generation_state.action])
+                else:
+                    self.durable_context.actions[0].append(generation_state.action)
+
+            elif isinstance(generation_state, TaskSet):
+                if self.durable_context._replay_schema == ReplaySchema.V1:
+                    self.durable_context.actions.append(generation_state.actions)
+                else:
+                    self.durable_context.actions[0].append(generation_state.actions)
```

Comment:
Do we even generate TaskSet even more in V2?

Reply:
We do! It's just its corresponding `actions` filed that is now represented with either a `WhenAllAction` or `WhenAnyAction`.

I did originally think of removing the `TaskSet` class with this PR, but that broke far too many unit tests to be viable at the moment. It is high on my priority list though